File: Java /Kode Program/Petshop.java
import java.util.Formatter; // Digunakan untuk pemformatan teks dalam output tabel

public class Petshop {
    // Atribut kelas untuk menyimpan data produk
    private int ID;          // ID produk
    private String Nama;     // Nama produk
    private String Kategori; // Kategori produk
    private int Harga;       // Harga produk

    // Konstruktor default (tanpa parameter)
    public Petshop() {
        this.ID = 0;
        this.Nama = "";
        this.Kategori = "";
        this.Harga = 0;
    }

    // Konstruktor dengan parameter (untuk inisialisasi objek langsung dengan nilai tertentu)
    public Petshop(int id, String nama, String kategori, int harga) {
        this.ID = id;
        this.Nama = nama;
        this.Kategori = kategori;
        this.Harga = harga;
    }

    // Setter methods (untuk mengubah nilai atribut secara individu)
    public void Set_ID(int id) {
        this.ID = id;
    }

    public void Set_Nama(String nama) {
        this.Nama = nama;
    }

    public void Set_Kategori(String kategori) {
        this.Kategori = kategori;
    }

    public void Set_Harga(int harga) {
        this.Harga = harga;
    }

    // Getter methods (untuk mengambil nilai atribut secara individu)
    public int Get_ID() {
        return this.ID;
    }

    public String Get_Nama() {
        return this.Nama;
    }

    public String Get_Kategori() {
        return this.Kategori;
    }

    public int Get_Harga() {
        return this.Harga;
    }

    // Menampilkan header tabel untuk daftar produk
    public static void tampilHeader() {
        System.out.println("+----+---------------------+----------------+-------------+");
        System.out.println("| ID | Nama                | Kategori       | Harga       |");
        System.out.println("+----+---------------------+----------------+-------------+");
    }

    // Menampilkan footer tabel (garis bawah)
    public static void tampilFooter() {
        System.out.println("+----+---------------------+----------------+-------------+");
    }

    // Menampilkan satu baris data produk dalam format tabel
    public void tampilBaris(int id, String nama, String kategori, int harga) {
        System.out.printf("| %2d | %-19s | %-14s | Rp. %7d |\n", id, nama, kategori, harga);
    }
}

File: Petshop.cpp
#include <iostream>
#include <string>
#include <iomanip>  

using namespace std;

class Petshop {
private:
    int ID;           // ID produk
    string Nama;      // Nama produk
    string Kategori;  // Kategori produk
    int Harga;        // Harga produk

public:
    // Konstruktor default
    Petshop() {
        ID = 0;
        Nama = "";
        Kategori = "";
        Harga = 0;
    }

    // Konstruktor dengan parameter untuk inisialisasi objek dengan data awal
    Petshop(int id, string nama, string kategori, int harga) {
        ID = id;
        Nama = nama;
        Kategori = kategori;
        Harga = harga;
    }

    // Setter untuk mengubah ID produk
    void Set_ID(int id) {
        ID = id;
    }

    // Setter untuk mengubah nama produk
    void Set_Nama(string nama) {
        Nama = nama;
    }

    // Setter untuk mengubah kategori produk
    void Set_Kategori(string kategori) {
        Kategori = kategori;
    }
    // Setter untuk mengubah harga produk
    void Set_Harga(int harga) {
        Harga = harga;
    }
    
    // Getter untuk mengambil ID produk
    int Get_ID() {
        return ID;
    }

    // Getter untuk mengambil nama produk
    string Get_Nama() {  
        return Nama;
    }

    // Getter untuk mengambil kategori produk
    string Get_Kategori() {  
        return Kategori;
    }

    // Getter untuk mengambil harga produk
    int Get_Harga() {
        return Harga;
    }

    // Fungsi untuk menampilkan header tabel
    void tampilHeader() {
        cout << "+----+---------------------+----------------+-------------+" << endl;
        cout << "| ID | Nama                | Kategori       | Harga       |" << endl;
        cout << "+----+---------------------+----------------+-------------+" << endl;
    }

    // Fungsi untuk menampilkan footer tabel
    void tampilFooter() {
        cout << "+----+---------------------+----------------+-------------+" << endl;
    }

    // Fungsi untuk menampilkan satu baris data produk dalam format tabel
    void tampilBaris(int id, string nama, string kategori, int harga) {
        // Batasi panjang teks agar tidak merusak tampilan tabel
        if (nama.length() > 20) nama = nama.substr(0, 17) + "...";
        if (kategori.length() > 15) kategori = kategori.substr(0, 12) + "...";

        cout << "| " << left << setw(2) << id  << " "  // Kolom ID
             << "| " << setw(20) << nama  // Kolom Nama (20 karakter)
             << "| " << setw(15) << kategori  // Kolom Kategori (15 karakter)
             << "| Rp " << right << setw(8) << harga  << " |" << endl;  // Kolom Harga
    }

    // Destruktor untuk membersihkan memori jika diperlukan
    ~Petshop() {}
};
